drafts/zip-str4d-sapling-anchor.rst

```diff
@@ -0,0 +1,110 @@
+::
+
+  ZIP: XXX
+  Title: Sapling Anchor Deduplication within Transactions
+  Owners: Jack Grigg <jack@z.cash>
+  Status: Draft
+  Category: Consensus
+  Created: 2019-03-27
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MAY" in this document are to be interpreted as described in
+RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in ZIP 200
+[#zip-0200]_.
+
+The term "Sapling" in this document is to be interpreted as described in ZIP 205
+[#zip-0205]_.
+
+Abstract
+========
+
+This proposal defines a modification to the transaction format whereby a single Sapling
+anchor is used for all Sapling spends. This change removes a potential implementation
+fingerprint, and reduces the size of Sapling transactions within the block chain.
+
+
+Motivation
+==========
+
+The Sapling network upgrade [#zip-0205]_ introduced new shielded inputs (spends) and
+outputs. Each spend proves the existence of the note being spent by including the anchor
+of a Merkle tree that contains the note's commitment, and proving in zero knowledge the
+existence of a path from the commitment to the anchor (a witness). Valid anchors
+correspond to the state of the Sapling commitment tree after each block in the chain.
+
+The choice of anchor leaks information about the note being spent, namely that the note
+was created no later than the anchor's block height. This is an unavoidable outcome of the
+Zcash design, and the least information it is possible to leak about a note being spent.
+However, the Sapling v4 transaction format [#v4-tx]_ includes a separate anchor for each
+Sapling spend, and thus it is possible to leak additional information by using different
+anchors for different notes. The anchor selection choices could also be used as a
+fingerprint to identify transactions created by particular wallet implementations,
+reducing the privacy set.
+
+Modifying the transaction format to have a single Sapling anchor field, instead of one
+field per Sapling spend, removes the ability (within the new transaction format version)
+to create transactions with this fingerprint. It also reduces the size of the transaction,
+costing 32 bytes per transaction instead of 32 bytes per spend.
+
+
+Specification
+=============
+
+A new transaction format is defined, identical to the Sapling v4 transaction format
+except for two changes:
+
+- The ``anchor`` field in ``SpendDescription`` is removed.
+- A new field ``saplingAnchor`` is placed between ``vShieldedOutput`` and ``vJoinSplit``,
+  if and only if ``vShieldedSpend`` is not empty.
+
+Consensus rules that previously applied to individual ``anchor`` entries MUST be applied
+to ``saplingAnchor``.
+
+TODO: If this is the only ZIP updating the transaction format in a NU, specify the full
```

Comment:
To me, "an NU" seems preferable to "a NU" since NU is (by the most common definitions) an initialism (one pronounces the individual letters), not an acronym (pronounceable as a word).

Reply:
Fair point, though in actual document usage it should always be `network upgrade`. This TODO will be removed from the eventual ZIP.